### ISSUE TITLE
Suggested fixes for FAQ 6 re seeing bounced invites

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe-old#694]
These are my suggestions for tweaks to reduce the number of people having trouble with FAQ 6: "Can I see if any of the email addresses I invited to a group or topic bounced?"

(1) "Pending invites" help button

I suggest adding a little "Pending invites" help button, as per the mockup below.

The suggested text for this help button would be:

"The members listed below as “Pending” are those who have been sent email invites but not yet accepted. 

However, if the email address ‘bounced’ (i.e. it was not a correct email address, and the invite was therefore not able to be delivered) it will still be listed here as “Pending”.  

Need more guidance? See Q6 in our FAQs (hyperlinked 'FAQs')."   

![image](https://user-images.githubusercontent.com/61320526/116236902-e3c4f680-a75f-11eb-808f-07eb822daa8d.png)

(2) Adding a "Guidance" help button when the user is entering emails

I suggest adding a little "Guidance" help button, as per the mockup below.

![image](https://user-images.githubusercontent.com/61320526/116237289-52a24f80-a760-11eb-909c-e0ab68b7c5f7.png)

**The suggested text for this help button for Groups invites would be:**

"We recommend posting at least one topic in the group before inviting users, as being invited to an empty group can cause confusion.
 
Multiple email addresses can be invited in one go, simply by separating them each with a space or comma.

Please note, if you enter an email address which does not actually exist (e.g. you type it wrong) the system will still consider it sent, and mark it as "Pending" in the group's user list. 

So before inviting users, make sure that the email addresses you have entered are correct and in use. 

Please also check that you have entered all email addresses in the correct format (i.e. “hello@email” rather than “hello@email.com”). Otherwise, the system will not accept it, and the invite will not be sent. 

Need more guidance? See Q6 and Q8 in our FAQs (hyperlinked 'FAQs')."   

**The suggested text for this help button for Topics invites would be:**

"Multiple email addresses can be invited in one go, simply by separating them each with a space or comma.

Please note, if you enter an email address which does not actually exist (e.g. you type it wrong) the system will still consider it sent, and mark it as "Pending" in the topic's user list. 

So before inviting users, make sure that the email addresses you have entered are correct and in use. 

Please also check that you have entered all email addresses in the correct format (i.e. “hello@email” rather than “hello@email.com”). Otherwise, the system will not accept it, and the invite will not be sent. 

Need more guidance? See Q6 and Q8 in our FAQs (hyperlinked 'FAQs')."   

(3) Allowing inviters to see the email addresses they invited to in the user list:

I suggest it should be possible for the person who sent the invites to see the full email address that was submitted, not just the username of the email. Then they can check retrospectively that they have sent the invites to the correct email address, if they have some that are pending. 

However, I also see there is a similar request here that would also solve this problem, and there may be a "tidier" way to do it: https://github.com/citizenos/citizenos-fe/issues/670

![image](https://user-images.githubusercontent.com/61320526/116239852-70bd7f00-a763-11eb-9e43-c324aa50690e.png)

(4) A way for the system to recognise which emails have bounced?

I assume it isn't possible for the system to recognise which emails have bounced, but if there is then we could consider how to make this apparent to the person who invited them.

**Comment by loorm** [comment]
Triage 36. This needs further thinking and discussion, also splitting up into separate issues. @kevincrepin input is also appretiated.

**Comment by kevincrepin** [comment]
Will also check this tomorrow.

**Comment by kevincrepin** [comment]
Hey,
We actually had designs in place to show pending invites and deal with some – if not maybe all – cases. But I couldn't find them on Invision and I don't know or remember in what context they were shared then... But I uploaded them now

Pending invite indication: https://projects.invisionapp.com/d/main#/console/9829159/450863948/preview
Revoking invitation: https://projects.invisionapp.com/d/main#/console/9829159/450863949/preview 
Expired invitation: https://projects.invisionapp.com/d/main#/console/9829159/450863951/preview

This doesn't cover all the cases Becca outlined of course and I think those ideas have merit.

**Comment by BeccaMelhuish** [comment]
I've now separated out this issue into two separate issues, and moved what was previously suggestion No. 3 from my original comment above to the new issue: #704 "Flagging invited emails that are in the wrong format".

**Comment by BeccaMelhuish** [comment]
@kevincrepin re the Invision designs you shared, I feel like they're not quite addressing the same issue, as the issue in this case is more that admins don't know whether their pending invites went to a valid email address or just "bounced". And they have no way of checking the email address it was sent to. So the idea would be to make the email address used visible to the "inviter", so that they can check it is a valide one. 

An update:
We (@tiblu @ilmartyrk, meeting on 06.05) discussed this and the main issue seems to be, as per related issue #670, it would be technically very difficult to only show the email addresses used to the "inviter" rather than to all admins. So other than the "guidance" button idea above, the others are on hold for now until we have legal advice on showing the email addresses to all admins.

**Comment by BeccaMelhuish** [comment]
Much of this has been resolved in #704 now ("Flagging invited emails that are in the wrong format"), or can easily be resolved when we have the Help icons ready (I'll re-write the help text then, depending on where we are with the improvements at that stage).

But one thing outstanding is the problem of whether it might be possible for the system to recognise bounced emails (suggestion (4)), and flag them up to the admin? When we discussed in our meeting (the April one) we said this would need some more consideration on Github - @ilmartyrk what do you think about this? Or anyone else of course :)

**Comment by BeccaMelhuish** [comment]
And "(3) Allowing inviters to see the email addresses they invited" in the user list still needs consideration too. I believe awaiting legal input from Kati in #670? 

I know Mikk was saying this could cause a lot of issues with permissions, if only allowing the person who actually invited the users to see the emails, so could be hard to put into practice. But would help the admin to be able to go back and check they entered the right addresses. Less helpful than (4) above though - that would be the ideal for this particular case.

**Comment by anettlinno** [comment]
If I understand correctly @BeccaMelhuish then _re:_ _(4) A way for the system to recognize which emails have bounced?_ is yet unsolved and other three issues are either solved or in action? 

What if bounced emails would be displayed on users list the same way as we show "pending invite"? It could say "bounced" or "unsend" email and admin users can see the email addresses (after we solve issue 670) and check the users list after entering all the emails. Both pending invites and bounced invites could be displayed either in the beginning or in the end of the list. 
But yes, as you pointed out, Becca then the question is to @ilmartyrk if the system is able to recognize bounced emails.

**Comment by anettlinno** [comment]
Triage 59. Our system doesn´t recognize bounced emails right now. We need to investigate possible solution. There is stats API that could send us back bounced emails, but needs further research about how to implement it in our system.
Approx. investigation time 1 hour. Sending this To do list.

**Comment by ilmartyrk** [comment]
Did some research and low scale testing. We can use https://www.npmjs.com/package/mailgun.js#suppressions to get list of e-mails that have bounced. There is also option to do by e-mail address checks to know if sending the e-mail has bounced or not.

To create solution for this issue, I can see 3 ways. 
1) Create automated check that runs after x-time and checks if any invites have bounced. This will get the list from API and add a marker to our DB for each bounced invite, so that we can display this info to admins
2) We implement e-mail based check that will go over each e-mail and then add the marker if succeeded or bounced
3) We combine both solutions.

**Comment by anettlinno** [comment]
Triage 69. We decided to go with the first provided solution, BUT it needs a User journey description as there are some issues arising: 

- If the system will send out the list of all bounced emails then where we will show it to the User? We discussed that we should have separate tab for that under "Manage participants". BUT how can User find bounced emails from there if one doesn´t know the platform too well? This needs very good UX.
- Can our system push notification for the User right after User has entered email addresses (and our system identified bounced email) and if yes, then where and how?

@BeccaMelhuish @kevincrepin your thoughts and ideas are very welcome :) 

Sending this issue to In prep for further UX improvement.
